src/Features/Core/Portable/CodeRefactoringHelpers.cs

```diff
@@ -30,45 +30,66 @@ internal static class CodeRefactoringHelpers
             }
 
+            // Selection or node is empty -> can't be underselected
             if (selection.IsEmpty || node.Span.IsEmpty)
             {
                 return false;
             }
 
-            // If selection is larger than node.Span -> can't be underselecting
+            // Selection is larger than node.Span -> can't be underselecting
             if (selection.Contains(node.Span))
             {
                 return false;
             }
 
-            // Only precisely one token is selected -> treat is as empty selection -> not under-selected.
-            // The rationale is that if a only one Token is selected then the selection wasn't about
-            // precisely getting the one node and nothing else & therefore we should treat it as empty
-            // selection.
-            var selectionStartToken = node.FindToken(selection.Start);
-            if (selection.IsAround(selectionStartToken))
+            // Selection doesn't intersect node -> can't be underselecting.
+            // RATIONALE: If there's no intersection then we got the node in some other way, e.g. 
+            // extracting it after user selected `;` at the end of an expression statement 
+            // `foo()[|;|]` for `foo()` node.
+            if (!node.FullSpan.OverlapsWith(selection))
             {
                 return false;
             }
 
+            // Only precisely one token of the node is selected -> treat is as empty selection -> not 
+            // underselected. The rationale is that if only one Token is selected then the selection 
+            // wasn't about precisely getting the one node and nothing else & therefore we should treat 
+            // it as empty selection.
+            if (node.FullSpan.Contains(selection.Start))
+            {
+                var selectionStartToken = node.FindToken(selection.Start);
+                if (selection.IsAround(selectionStartToken))
+                {
+                    return false;
+                }
+            }
+
             var beginningNode = node.FindToken(node.Span.Start).Parent;
             var endNode = node.FindToken(node.Span.End - 1).Parent;
 
             // Node is underselected if either the first (lowest) child doesn't contain start of selection
             // of the last child doesn't intersect with the end.
+
+            // Node is underselected if either the first (lowest) child ends before the selection has started
+            // or the last child starts before the selection ends (i.e. one of them is completely on the outside of selection).
             // It's a crude heuristic but it allows omitting parts of nodes or trivial tokens from the beginning/end 
             // but fires up e.g.: `1 + [|2 + 3|]`.
-            return !beginningNode.Span.IntersectsWith(selection.Start) || !endNode.Span.IntersectsWith(selection.End - 1);
+            return beginningNode.Span.End <= selection.Start || endNode.Span.Start > selection.End - 1;
```

Comment:
instead of `> .End - 1` why not `>= End`?

Reply:
Artifact of change. That said, seems ?more explicit? to me?

Reply:
Wouldn't fix it normally but want to retrigger tests because of flaky Vb test anyway. 
(Not having permissions to just rerun tests is serious annoyance).